compiler/ast-passes/src/canonicalization/canonicalizer.rs

```diff
@@ -371,7 +371,7 @@ impl Canonicalizer {
             }
             Statement::Definition(definition) => {
                 let value = self.canonicalize_expression(&definition.value);
-                let type_ = self.canonicalize_self_type(definition.type_.as_ref());
+                let type_ = self.canonicalize_self_type(Some(&definition.type_)).unwrap();
```

Comment:
Is it possible to remove the redundancy of creating an `Option` and then calling `.unwrap()` on it here?

Reply:
Changes were made, but had to make some other changes like the static access nodes, as well as functions now always have a return type at parse time. i.e. a function with no return type is set to have a return type of `()` at parse time rather than canonicalization. @acoglio hope this is okay.

Reply:
I think this should be fine. Does this change slightly the JSON AST, for @bendyarm's converter?

Reply:
@acoglio it would make it for both static accesses and function output fields are never `null` but at least the same JSON that would normally appear for the `()` type.